proposals/semi-auto-properties.md

```diff
@@ -1,14 +1,325 @@
+
+```cs
+{ get => field ?? parent.AmbientValue; set; }
+```
+
+Both accessors may be full accessors with either one or both making use of `field`:
+
+```cs
+{ get => field; set => field = value; }
+```
+
+```cs
+{ get => field; set => throw new InvalidOperationException(); }
+```
+
+```cs
+{ get => overriddenValue; set => field = value; }
+```
 
-In these cases by now you always have to create an instance field and write the whole property yourself.  This not only adds a fair amount of code, but it also leaks the `field` into the rest of the type's scope, when it is often desirable to only have it be available to the bodies of the accessors.
+```cs
+{
+    get;
+    set
+    {
+        if (field == value) return;
+        field = value;
+        OnXyzChanged();
+    }
+}
+```
+
+Expression-bodied properties and properties with only a `get` accessor may also use `field`:
+
+```cs
+public string LazilyComputed => field ??= Compute();
+```
+
+```cs
+public string LazilyComputed { get => field ??= Compute(); }
+```
+
+As with auto properties, a setter that uses a backing field is disallowed when there is no getter. This restriction could be loosened in the future to allow the setter to do something only in response to changes, by comparing `value` to `field` (see open questions).
+
+```cs
+// ❌ Error, will not compile
+{ set => field = value; }
+```
+
+### Breaking changes
+
+The existence of the `field` contextual keyword within property accessor bodies is a potentially breaking change, proposed as part of a larger [Breaking Changes](https://github.com/dotnet/csharplang/issues/7964) feature.
+
+Since `field` is a keyword and not an identifier, it can only be "shadowed" by an identifier using the normal keyword-escaping route: `@field`. All identifiers named `field` declared within property accessor bodies can safeguard against breaks when upgrading from C# versions prior to 13 by adding the initial `@`.
+
+### Field-targeted attributes
+
+As with auto properties, any property that uses a backing field in one of its accessors will be able to use field-targeted attributes:
+
+```cs
+[field: Xyz]
+public string Name => field ??= Compute();
+
+[field: Xyz]
+public string Name { get => field; set => field = value; }
+```
+
+A field-targeted attribute will remain invalid unless an accessor uses a backing field:
+
+```cs
+// ❌ Error, will not compile
+[field: Xyz]
+public string Name => Compute();
+```
+
+### Property initializers
+
+Properties with initializers may use `field`. The backing field is directly initialized rather than the setter being called ([LDM decision](https://github.com/dotnet/csharplang/blob/main/meetings/2022/LDM-2022-03-02.md#open-questions-in-field)).
+
+Calling a setter for an initializer is not an option; initializers are processed before calling base constructors, and it is illegal to call any instance method before the base constructor is called. This is also important for default initialization/definite assignment of structs.
```

Comment:
This would also not work in any sensible fashion in the presence of virtual properties.